packages/fxa-content-server/app/scripts/views/mixins/brand-messaging-mixin.js

```diff
@@ -0,0 +1,85 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+/**
+ * A mixin for brand in product messaging
+ */
+import Template from 'templates/partial/brand-messaging.mustache';
+import Storage from '../../lib/storage';
+
+const bannerClosedLocalStorageKey = 'fxa_disable_brand_banner';
+
+/**
+ * Creates the brand banner mixin.
+ *
+ * @param {Object} [config={}]
+ *   @param {Boolean} [config.required=false] Is the input field required on form submit?
+ * @returns {Function}
+ */
+const Mixin = {
+  initialize(options) {
+    this.mode = options.config.brandMessagingMode;
+  },
+
+  events: {
+    [`click #close-brand-banner`]: 'onBrandBannerClose',
+    [`click .brand-learn-more`]: 'onBrandLearnMoreClick',
+  },
+
+  beforeRender() {
+    // Check to see if the user has cleared the brand messaging banner. If so,
+    // we don't want to keep showing them this messaging
+    this.disableBanner = Storage.factory('localStorage', this.window).get(
+      `${bannerClosedLocalStorageKey}_${this.mode}`
+    );
```

Comment:
Does this mean someone could see the messaging twice, since the message dismissal is per mode?

Reply:
Yeah they are two different messages. This wasn't behavior wasn't defined, so this seemed better to me.